cosmos/models/Workflow.py

```diff
@@ -591,15 +600,20 @@ def handle_exits(workflow, do_atexit=True):
     if do_atexit:
         @atexit.register
         def cleanup_check():
+            workflow_terminated = False
             try:
                 try:
                     if workflow.status in [WorkflowStatus.running, WorkflowStatus.failed_but_running]:
                         workflow.log.error('%s still running when atexit() was called, terminating' % workflow)
                         workflow.terminate(due_to_failure=True)
+                        workflow_terminated = True
                 except SQLAlchemyError:
                     workflow.log.error('%s Unknown status when atexit() was called (sql error), terminating' % workflow)
                     workflow.terminate(due_to_failure=True)
+                    workflow_terminated = True
             finally:
+                if not workflow_terminated:
+                    workflow.terminate(due_to_failure=False, is_cleanup=True)
```

Comment:
would it be better not to overload workflow.terimate() and have a second function called workflow.cleanup()?  if it's a DRM specific cleanup, maybe that cleanup should be specified in the DRM class?

Reply:
I think we need to ensure via flow control that we haven't killed these tasks before.